src/status_im/ui/components/search_input/view.cljs

```diff
@@ -3,12 +3,15 @@
             [quo.design-system.colors :as colors]
             [quo2.foundations.colors :as quo2.colors]
             [reagent.core :as reagent]
-            [utils.i18n :as i18n]))
+            [utils.i18n :as i18n]
+            [quo2.theme :as quo.theme]
+  ))
 
 (defn search-input
   [{:keys [search-active?]}]
   (let [input-ref      (atom nil)
-        search-active? (or search-active? (reagent/atom nil))]
+        search-active? (or search-active? (reagent/atom nil))
+        theme          (quo.theme/get-theme)]
     (fn [{:keys [on-focus on-change on-blur on-cancel search-filter auto-focus]}]
       [quo/text-input
        {:placeholder         (i18n/label :t/search)
```

Comment:
I think we should wrap  this `search-input` in our theme HOC, or move the `theme` binding to the inner function.

If the app theme is set to change according to user's OS and the user changes the theme while being in a screen where this component is mounted, the `theme` value won't be recomputed until the user goes back and forth to this screen (i.e. remounts this component).

Just to be a little more clear:
```clojure
(defn search-input []
  (let [some  'bindings
        ;; Recomputed only once, it doesn't matter theme changes
        ;; it requires component remount to take effect
        theme (quo.theme/get-theme)]
    (fn [ ,,, ]
    [:<>
     ,,,
     ])))

(defn search-input []
  (let [some 'bindings]
    (fn [ ,,, ]
      ;; Recomputed on every re-render, if theme changes it will change instantaneously
      (let [theme (quo.theme/get-theme)]
        [:<>
         ,,,
         ]))))

```

Reply:
sure, I will adjust. I didn't pay much attention here as this is in status-im namespace and hopefully it is nuked soon anyway, but you are right. it's better to implement correctly 👍

Reply:
oh yes, that's true, if it is not being used I think it doesn't matter, but yes, let's try to implement it in the right way :+1: